### PR TITLE
plugins.rtpplay: add schema and fix HLS URL

### PR DESCRIPTION
**Pull request by mvaladas**
RTP Play Plugin was failing for a couple of weeks now. The website changed structure and the URL parsing needed updating.

### CODE REVIEW

File: src/streamlink/plugins/rtpplay.py [hunk 1]
```diff
@@ -25,7 +26,8 @@ def _get_streams(self):
             log.error("Could not find _m3u8_re")
             return
 
-        hls_url = m.group("url")
+        hls_url = eval("\"\".join((" + m.group(1) + "))")
```
Comment (back-to) [inline src/streamlink/plugins/rtpplay.py:29]: don't use `eval`

**Review by back-to — CHANGES_REQUESTED** [review]

File: src/streamlink/plugins/rtpplay.py (hunk 1 above)
Comment (bastimeyer) [reply src/streamlink/plugins/rtpplay.py:29]: Get the array and its content without the `.join()` call and parse it as JSON. Then do the array to string conversion and decoding afterwards in the plugin logic.

Comment (bastimeyer) [reply src/streamlink/plugins/rtpplay.py:29]: This does obviously only work as long as the site doesn't add more "code obfuscation" which would require a JS runtime, like for example dynamic string concatenation or function calls, etc. Some sites do this so their sources can't be statically parsed, but this seems simple enough.

**Comment by bastimeyer** [comment]
```
$ streamlink -l debug https://www.rtp.pt/play/direto/rtp1
[cli][debug] OS:         Linux-5.11.6-2-git-x86_64-with-glibc2.33
[cli][debug] Python:     3.9.2
[cli][debug] Streamlink: 2.0.0+57.gd8e980e
[cli][debug] Requests(2.25.1), Socks(1.7.1), Websocket(0.57.0)
[cli][info] Found matching plugin rtpplay for URL https://www.rtp.pt/play/direto/rtp1
[utils.l10n][debug] Language code: en_US
Available streams: 640k (worst), 1290k, 2540k (best)
```

```
$ streamlink -l debug https://www.rtp.pt/play/direto/antena1
[cli][debug] OS:         Linux-5.11.6-2-git-x86_64-with-glibc2.33
[cli][debug] Python:     3.9.2
[cli][debug] Streamlink: 2.0.0+57.gd8e980e
[cli][debug] Requests(2.25.1), Socks(1.7.1), Websocket(0.57.0)
[cli][info] Found matching plugin rtpplay for URL https://www.rtp.pt/play/direto/antena1
[utils.l10n][debug] Language code: en_US
Available streams: 170k (worst, best)
```

**Pull request merged**